src/BlocklyInterface/BlocklyEditor.tsx

```diff
@@ -130,6 +133,25 @@ export const BlocklyEditor: FunctionComponent = () => {
     }
   }, [highlightedBlock, currentBlockSelection, executing]);
 
+  // show/minimize the toolbox. Blockly does not allow the manipulation of the toolbox
+  // in any way except updating the xml definition of it.
+  useEffect(() => {
+    if (blocklyRef.current) {
+      const workspace = Blockly.getMainWorkspace() as WorkspaceSvg;
+      const blocklyXml = showToolbox ? getToolbox() : getEmptyToolbox();
```

Comment:
Would it make sense to actually have the redux slice hold the current toolbox and for this component to simply display whatever that toolbox is?

I could see in the future we'd have multiple toolboxes which seems like a piece of business logic / state. 

I'd still keep the exact same action you made and hold whether it's collapsed or not in redux still.

Reply:
blockly allows only ONE toolbox :-( 
To have multiple ones you have to change this one to whatever it should display right now. In fact it's just an illusion to the user that he has multiple toolboxes (or collapsed/expanded)